server/src/main/scala/org/labrad/browser/NodeApi.scala

```diff
@@ -63,12 +63,16 @@ class NodeApi(cxn: LabradConnection)(implicit ec: ExecutionContext) extends Logg
     }
   }
 
-  def autostartAdd(node: String, server: String): Unit = {
-    cxn.to(node).call("autostart_add", Str(server))
+  def autostartAdd(node: String, server: String): Future[Unit] = {
+    async {
+      cxn.to(node).call("autostart_add", Str(server))
```

Comment:
Actually just saying async here doesn't help. The issue is that `.call` already returns a `Future`, so you need to await it inside the async block, or just return the Future directly. I guess it's actually a `Future[Data]`, and we can't send arbitrary labrad data over jsonrpc (alas), so would have to drop the resulting value. Either one of these would do the trick:
```scala
def autostartAdd(...): Future[Unit] = {
  cxn.to(node).call(...).map { _ => () } // discard the Data
}

def autostartAdd(...): Future[Unit] = {
  async {
    await { cxn.to(node).call(...) }
  }
}
```